### ISSUE TITLE
Cannot build MPICH

### ISSUE THREAD

**Issue by eschnett** [JuliaPackaging/Yggdrasil#4997]
When I build MPICH for macOS, the created library `libmpifort.dylib` cannot be found.

```sh
julia --color=yes build_tarballs.jl --debug --verbose x86_64-apple-darwin-libgfortran5
```

The error is likely:
```
[ Info: Checking shared library lib/libmpi.12.dylib
[ Info: lib/libmpi.12.dylib already has SONAME "@rpath/libmpi.12.dylib"
[ Info: Checking shared library lib/libmpicxx.12.dylib
[ Info: lib/libmpicxx.12.dylib already has SONAME "@rpath/libmpicxx.12.dylib"
[ Info: Checking shared library lib/libmpifort.12.dylib
ERROR: could not load library "/Users/eschnett/src/Yggdrasil/M/MPICH/build/x86_64-apple-darwin-libgfortran5/h5Z8FDHP/x86_64-apple-darwin14-libgfortran5-cxx11/destdir/lib/libmpifort.12.dylib"
dlopen(/Users/eschnett/src/Yggdrasil/M/MPICH/build/x86_64-apple-darwin-libgfortran5/h5Z8FDHP/x86_64-apple-darwin14-libgfortran5-cxx11/destdir/lib/libmpifort.12.dylib, 0x0001): Library not loaded: /workspace/destdir/x86_64-apple-darwin14/lib/libgcc_s.1.1.dylib
  Referenced from: /Users/eschnett/src/Yggdrasil/M/MPICH/build/x86_64-apple-darwin-libgfortran5/h5Z8FDHP/x86_64-apple-darwin14-libgfortran5-cxx11/artifacts/dfbcd978b9dc8dcb06dbc1eff782d5ce8b54d80a/lib/libgcc_s.1.dylib
  Reason: tried: '/workspace/destdir/x86_64-apple-darwin14/lib/libgcc_s.1.1.dylib' (no such file), '/usr/local/lib/libgcc_s.1.1.dylib' (no such file), '/usr/lib/libgcc_s.1.1.dylib' (no such file)
┌ Warning: lib/libmpifort.12.dylib cannot be dlopen()'ed
└ @ BinaryBuilder.Auditor ~/.julia/packages/BinaryBuilder/gIX21/src/Auditor.jl:171
[ Info: lib/libmpifort.12.dylib already has SONAME "@rpath/libmpifort.12.dylib"
[ Info: Checking shared library lib/libpmpi.12.dylib
ERROR: could not load library "/Users/eschnett/src/Yggdrasil/M/MPICH/build/x86_64-apple-darwin-libgfortran5/h5Z8FDHP/x86_64-apple-darwin14-libgfortran5-cxx11/destdir/lib/libpmpi.12.dylib"
dlopen(/Users/eschnett/src/Yggdrasil/M/MPICH/build/x86_64-apple-darwin-libgfortran5/h5Z8FDHP/x86_64-apple-darwin14-libgfortran5-cxx11/destdir/lib/libpmpi.12.dylib, 0x0001): symbol not found in flat namespace '_ADIOI_Datarep_head'
┌ Warning: lib/libpmpi.12.dylib cannot be dlopen()'ed
└ @ BinaryBuilder.Auditor ~/.julia/packages/BinaryBuilder/gIX21/src/Auditor.jl:171
```

Looking for `libgcc_s.*` via `find` (from outside the Docker container) shows:
```sh
$ find . -name 'libgcc_s.*'
./build/x86_64-apple-darwin-libgfortran5/h5Z8FDHP/x86_64-apple-darwin14-libgfortran5-cxx11/artifacts/dfbcd978b9dc8dcb06dbc1eff782d5ce8b54d80a/lib/libgcc_s.1.1.dylib
./build/x86_64-apple-darwin-libgfortran5/h5Z8FDHP/x86_64-apple-darwin14-libgfortran5-cxx11/artifacts/dfbcd978b9dc8dcb06dbc1eff782d5ce8b54d80a/lib/libgcc_s.1.dylib
./build/x86_64-apple-darwin-libgfortran5/h5Z8FDHP/x86_64-apple-darwin14-libgfortran5-cxx11/destdir/lib/libgcc_s.1.1.dylib
./build/x86_64-apple-darwin-libgfortran5/h5Z8FDHP/x86_64-apple-darwin14-libgfortran5-cxx11/destdir/lib/libgcc_s.1.dylib
./build/x86_64-apple-darwin-libgfortran5/iiGn44Ox/x86_64-apple-darwin14-libgfortran5-cxx11/artifacts/44220be3f55a7ad5e50238bfb1d9a76f4a8bdee1/lib/libgcc_s.1.dylib
./build/x86_64-apple-darwin-libgfortran5/iiGn44Ox/x86_64-apple-darwin14-libgfortran5-cxx11/destdir/lib/libgcc_s.1.dylib
```

Is this an rpath problem? Are the generated libraries pointing to absolute paths within the Docker image, and they cannot be resolved from the outside?

**Comment by giordano** [comment]
I guess the problem is that GCC 12 broke the ABI of `libgcc_s` for darwin: https://github.com/JuliaPackaging/Yggdrasil/pull/4985#issuecomment-1146668557.  My guess is that you should be able to work around this issue by building with `CompilerSupportLibraries_jll` `v0.5.2`.

To be sure, `libmpifort.12.dylib` doesn't require `libgcc_s` directly, right?  I assume that comes through libgfortran.

**Comment by giordano** [comment]
Looking at latest build of mpich:
```console
sandbox:${WORKSPACE} # otool -L *.dylib
libfmpich.dylib:
        @rpath/libmpifort.12.dylib (compatibility version 15.0.0, current version 15.2.0)
        @rpath/libmpi.12.dylib (compatibility version 15.0.0, current version 15.2.0)
        @rpath/libpmpi.12.dylib (compatibility version 15.0.0, current version 15.2.0)
        @rpath/libgfortran.5.dylib (compatibility version 6.0.0, current version 6.0.0)
        @rpath/libquadmath.0.dylib (compatibility version 1.0.0, current version 1.0.0)
        /usr/lib/libSystem.B.dylib (compatibility version 1.0.0, current version 1292.60.1)
libload_time_mpi_constants.dylib:
        @rpath/libload_time_mpi_constants.dylib (compatibility version 0.0.0, current version 0.0.0)
        @rpath/libmpi.12.dylib (compatibility version 15.0.0, current version 15.2.0)
        @rpath/libpmpi.12.dylib (compatibility version 15.0.0, current version 15.2.0)
        /usr/lib/libSystem.B.dylib (compatibility version 1.0.0, current version 1292.60.1)
libmpi.12.dylib:
        @rpath/libmpi.12.dylib (compatibility version 15.0.0, current version 15.2.0)
        @rpath/libpmpi.12.dylib (compatibility version 15.0.0, current version 15.2.0)
        /usr/lib/libSystem.B.dylib (compatibility version 1.0.0, current version 1292.60.1)
libmpi.dylib:
        @rpath/libmpi.12.dylib (compatibility version 15.0.0, current version 15.2.0)
        @rpath/libpmpi.12.dylib (compatibility version 15.0.0, current version 15.2.0)
        /usr/lib/libSystem.B.dylib (compatibility version 1.0.0, current version 1292.60.1)
libmpich.dylib:
        @rpath/libmpi.12.dylib (compatibility version 15.0.0, current version 15.2.0)
        @rpath/libpmpi.12.dylib (compatibility version 15.0.0, current version 15.2.0)
        /usr/lib/libSystem.B.dylib (compatibility version 1.0.0, current version 1292.60.1)
libmpichcxx.dylib:
        @rpath/libmpicxx.12.dylib (compatibility version 15.0.0, current version 15.2.0)
        @rpath/libmpi.12.dylib (compatibility version 15.0.0, current version 15.2.0)
        @rpath/libpmpi.12.dylib (compatibility version 15.0.0, current version 15.2.0)
        /usr/lib/libSystem.B.dylib (compatibility version 1.0.0, current version 1292.60.1)
        /usr/lib/libc++.1.dylib (compatibility version 1.0.0, current version 904.4.0)
libmpichf90.dylib:
        @rpath/libmpifort.12.dylib (compatibility version 15.0.0, current version 15.2.0)
        @rpath/libmpi.12.dylib (compatibility version 15.0.0, current version 15.2.0)
        @rpath/libpmpi.12.dylib (compatibility version 15.0.0, current version 15.2.0)
        @rpath/libgfortran.5.dylib (compatibility version 6.0.0, current version 6.0.0)
        @rpath/libquadmath.0.dylib (compatibility version 1.0.0, current version 1.0.0)
        /usr/lib/libSystem.B.dylib (compatibility version 1.0.0, current version 1292.60.1)
libmpicxx.12.dylib:
        @rpath/libmpicxx.12.dylib (compatibility version 15.0.0, current version 15.2.0)
        @rpath/libmpi.12.dylib (compatibility version 15.0.0, current version 15.2.0)
        @rpath/libpmpi.12.dylib (compatibility version 15.0.0, current version 15.2.0)
        /usr/lib/libSystem.B.dylib (compatibility version 1.0.0, current version 1292.60.1)
        /usr/lib/libc++.1.dylib (compatibility version 1.0.0, current version 904.4.0)
libmpicxx.dylib:
        @rpath/libmpicxx.12.dylib (compatibility version 15.0.0, current version 15.2.0)
        @rpath/libmpi.12.dylib (compatibility version 15.0.0, current version 15.2.0)
        @rpath/libpmpi.12.dylib (compatibility version 15.0.0, current version 15.2.0)
        /usr/lib/libSystem.B.dylib (compatibility version 1.0.0, current version 1292.60.1)
        /usr/lib/libc++.1.dylib (compatibility version 1.0.0, current version 904.4.0)
libmpifort.12.dylib:
        @rpath/libmpifort.12.dylib (compatibility version 15.0.0, current version 15.2.0)
        @rpath/libmpi.12.dylib (compatibility version 15.0.0, current version 15.2.0)
        @rpath/libpmpi.12.dylib (compatibility version 15.0.0, current version 15.2.0)
        @rpath/libgfortran.5.dylib (compatibility version 6.0.0, current version 6.0.0)
        @rpath/libquadmath.0.dylib (compatibility version 1.0.0, current version 1.0.0)
        /usr/lib/libSystem.B.dylib (compatibility version 1.0.0, current version 1292.60.1)
libmpifort.dylib:
        @rpath/libmpifort.12.dylib (compatibility version 15.0.0, current version 15.2.0)
        @rpath/libmpi.12.dylib (compatibility version 15.0.0, current version 15.2.0)
        @rpath/libpmpi.12.dylib (compatibility version 15.0.0, current version 15.2.0)
        @rpath/libgfortran.5.dylib (compatibility version 6.0.0, current version 6.0.0)
        @rpath/libquadmath.0.dylib (compatibility version 1.0.0, current version 1.0.0)
        /usr/lib/libSystem.B.dylib (compatibility version 1.0.0, current version 1292.60.1)
libmpl.dylib:
        @rpath/libmpi.12.dylib (compatibility version 15.0.0, current version 15.2.0)
        @rpath/libpmpi.12.dylib (compatibility version 15.0.0, current version 15.2.0)
        /usr/lib/libSystem.B.dylib (compatibility version 1.0.0, current version 1292.60.1)
libopa.dylib:
        @rpath/libmpi.12.dylib (compatibility version 15.0.0, current version 15.2.0)
        @rpath/libpmpi.12.dylib (compatibility version 15.0.0, current version 15.2.0)
        /usr/lib/libSystem.B.dylib (compatibility version 1.0.0, current version 1292.60.1)
libpmpi.12.dylib:
        @rpath/libpmpi.12.dylib (compatibility version 15.0.0, current version 15.2.0)
        /usr/lib/libSystem.B.dylib (compatibility version 1.0.0, current version 1292.60.1)
libpmpi.dylib:
        @rpath/libpmpi.12.dylib (compatibility version 15.0.0, current version 15.2.0)
        /usr/lib/libSystem.B.dylib (compatibility version 1.0.0, current version 1292.60.1)
```
so yeah, nothing depends directly on `libgcc_s`.  I don't think this would be a problem in practice, it's only during audit that the local libgfortran5 is probably unable to find `libgcc_s.1.1.dylib` because there is already `libgcc_s.1.dylib`.  At least so I hope :thinking:

**Comment by giordano** [comment]
This was presumably solved: #5009
